extensions/Bookalope/jsx/bookalope.jsx

```diff
@@ -193,3 +193,264 @@ function bookalopeCreateDocument(idmlFileName, bookId, bookflowId, betaHost) {
+		}
+		myAO.anchoredObjectSettings.properties = doc.anchoredObjectSettings.properties;
+		myAO.anchoredObjectSettings.anchoredPosition = AnchorPosition.anchored;
+		myAO.anchoredObjectSettings.pinPosition = false;
+		return myAO
+	}
+
+	//here the script starts
+    if (doc && doc.isValid) {
+    
+    	//checking if the document is saved
+    	if(!doc.saved || doc.modified) {
+			alert("Please save the file");
+			return false;
+		}
+		
+		//checking if all links exist
+		for (i = 0; i < doc.links.length; i++) {
+			var link = doc.links.item(i);
+			if(link.status == LinkStatus.LINK_INACCESSIBLE || link.status == LinkStatus.LINK_MISSING || link.status == LinkStatus.LINK_OUT_OF_DATE) {
+				alert("Please check links: some are not updated or missed");
+				return false;
+			}
+		}
+
+		//creating a temporary copy and opening it
+        tmp_path = app.createTemporaryCopy(doc.fullName);
+        //todo: switch to showingWindow=false
+        app.open(tmp_path, showingWindow=false);
+        doc = app.activeDocument;
+        
+		//step 1.1: add text condition and style
+		var stylePN = "bookalope-page-number";
+		var pages = doc.pages;
+		try {
+			myCharacterStyle = app.activeDocument.characterStyles.item(stylePN);
+			myName = myCharacterStyle.name;
+
+		} catch (myError) {
+			myCharacterStyle = app.activeDocument.characterStyles.add({ name: stylePN });
+		}
+		myCharacterStyle.pointSize = 0.1;
+		
+		//step 1.2: set page numbers – preferences
+		var w = new Window('palette', 'Adding page number references...');
+		w.pbar = w.add('progressbar', undefined, 0, pages.length);
+		w.pbar.preferredSize.width = 300;
+		w.show();
+		for (var i = 0; i < pages.length; i++) {
+			var myObjectList = new Array;
+			var items = pages[i].textFrames;
+			if (items.length !== 0) {
+				for (var j = 0; j < items.length; j++) {
+					if (items[j].parent.constructor.name === "Spread") {
+						if (items[j].contents !== "" && (items[j].nextTextFrame || items[j].previousTextFrame)) {
+							myObjectList.push(items[j]);
+						}
+					}
+				}
+				if (myObjectList.length !== 0) {
+					myObjectList.sort(function (a,b) { return (a.geometricBounds[0] < b.geometricBounds[0]) || (a.geometricBounds[0] == b.geometricBounds[0] && a.geometricBounds[1] < b.geometricBounds[1]) ? -1 : 1; } );
+					myTextFrame = myObjectList[0];
+					var myInsertionPoint = myTextFrame.insertionPoints.item(0),
+						myCharacterStyle = app.activeDocument.characterStyles.item(stylePN);
+
+					myInsertionPoint.contents = "" + pages[i].name;
+					myInsertionPoint.applyCharacterStyle(myCharacterStyle, true);
+				}
+			}
+			w.pbar.value = i;
+		}
+		
+		//step 2: dumpPastedImages – exports images that are embedded
+		var g = doc.allGraphics;
+		var outfolder = File(tmp_path).path;
+		for (var i = 0; i < g.length; i++) {
+			if (g[i].itemLink == null) {
+				image_file = File (uniqueName(outfolder+'/img', '.png'));
+				export_other (g[i], ExportFormat.PNG_FORMAT, image_file);
```

Comment:
@gregoriopellegrino, I can’t find the function implementation for `export_other()` — not in this PR and not in the [InDesign ExtendScript API](https://www.indesignjs.de/extendscriptAPI/indesign-latest/) 🤔

Reply:
Fixed :), I'm trying to test it locally, but I have difficulty activating it on InDesign...

Reply:
Thank you, @gregoriopellegrino!

> I have difficulty activating it on InDesign...

You mean to install the extension locally for InDesign?

Reply:
@gregoriopellegrino I accidentally dropped your commit when I force-pushed after a rebase — could you please commit that function again?

Reply:
Done!

Reply:
Thank you! ☺️